app/forms/support/edit_course_form.rb

```diff
@@ -0,0 +1,94 @@
+module Support
+  class EditCourseForm
+    include ActiveModel::Model
+    include ActiveModel::Validations
+
+    FIELDS = %i[
+      course_code name
+    ].freeze
+
+    attr_accessor(*FIELDS)
+    attr_accessor :day, :month, :year
+    validate :validate_start_date_format
+
+    def initialize(course)
+      @course = course
+
+      super(
+        course_code: @course.course_code,
+        name: @course.name,
+        day: @course.start_date&.day,
+        month: @course.start_date&.month,
+        year: @course.start_date&.year,
+      )
+    end
+
+    def save
+      valid?
+      course_valid?
+      return false unless errors.none?
+
+      @course.save
+    end
+
+    def course_valid?
+      assign_attributes_to_course
```

Comment:
I don't agree with this, assigning attributes is one thing, checking is it valid is another. why coupled?

Reply:
because attributes are coupled to the form not the course model. In order to check the validity of attr on the course before saving need to assign them